include/lsst/meas/extensions/photometryKron.h

```diff
@@ -150,6 +152,71 @@ class KronFluxAlgorithm : public base::SimpleAlgorithm {
     meas::base::SafeCentroidExtractor _centroidExtractor;
 };
 
+class KronAperture {
+public:
+    KronAperture(afw::geom::Point2D const& center, afw::geom::ellipses::BaseCore const& core) :
+        _center(center), _axes(core) {}
+
+    explicit KronAperture(afw::table::SourceRecord const& source) :
+        _center(afw::geom::Point2D(source.getX(), source.getY())), _axes(source.getShape()) {}
+
+    KronAperture(afw::table::SourceRecord const& reference, afw::geom::AffineTransform const& refToMeas,
+                 double radius) :
+        _center(refToMeas(reference.getCentroid())),
+        _axes(getKronAxes(reference.getShape(), refToMeas.getLinear(), radius))
+        {}
+
+    /// Accessors
+    double getX() const { return _center.getX(); }
+    double getY() const { return _center.getY(); }
+
+    afw::geom::Point2D const& getCenter() const { return _center; }
+
+    afw::geom::ellipses::Axes & getAxes() { return _axes; }
+
+    afw::geom::ellipses::Axes const& getAxes() const { return _axes; }
+
+    /// Determine the Kron Aperture from an image
+    ///
+    /// Determines the object Kron aperture, using the shape from source.getShape()
+    /// (e.g. SDSS's adaptive moments)
+    template<typename ImageT>
+    static PTR(KronAperture) determineRadius(
+        ImageT const& image,  ///< Image to measure
+        afw::geom::ellipses::Axes axes,  ///< Shape of aperture
+        afw::geom::Point2D const& center,   ///< Centre of source
+        KronFluxControl const& ctrl,  ///< control the algorithm
+        float *radiusForRadius   ///< radius used to estimate radius
```

Comment:
can you assure that this float pointer will not die or change somehow? I fear this might make things not thread safe if we go that way at some point. It is probably not too much of a penalty to just construct a float for this, or is there some other reason for this that I am missing?